lms/djangoapps/discussion/views.py

```diff
@@ -963,3 +991,13 @@ def get_divided_discussions(course, discussion_settings):
             divided_inline_discussions.append(divided_discussion_id)
 
     return divided_course_wide_discussions, divided_inline_discussions
+
+
+def _check_team_discussion_access(request, course, discussion_id):
+    """
+    Helper function to check if the discussion is visible to the user,
+    if the user is on a team, which has the discussion set to private.
+    """
+    user_is_course_staff = has_access(request.user, "staff", course)
+    if not user_is_course_staff and not team_api.discussion_visible_by_user(discussion_id, request.user):
```

Comment:
Nit: the logic in this check is little hard to read

Reply:
It's either `not(x or y)`, or `not x and not y`.
I just wanted to avoid the bracket.

Reply:
But as all good math teachers know, parentheses are your friend!